### PR TITLE
Core react logic for accounts

### PR DESCRIPTION
**Pull request by TimMikeladze**
This PR provides the core logic for individual react component libraries for accounts (material, semantic, bootstrap etc). It handles the generic aspects such rendering correct components based on the Accounts configuration, state management, tracking field values, errors, and submitting data. 

Pending #6 - need to update it to the latest version.

### CODE REVIEW

File: src/SignupFields.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          <AccountsClient.ui.EmailOptionalField {...emailProps({ form, fields })} />
+          <AccountsClient.ui.PasswordField {...passwordProps({ form, fields })} />
+          <AccountsClient.ui.PasswordConfirmField {...passwordConfirmProps({ form, fields })} />
+        </div>
+      );
+      break;
+    default:
+      break;
+  }
+  return jsx;
```
Comment (alexyuly) [inline src/SignupFields.js]: Style preference - `return` from each case rather than using `let`.

File: src/accountRoutes.js [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
-import AccountsClient from '@accounts/accounts';
+import { AccountsClient } from '@accounts/accounts';
+import FormTypes from './FormTypes';
 
-const accountRoutes = ({
-  component,
-}) =>
-  <Route component={component}>
-    <Route path={AccountsClient.ui._options.loginPath} component={AccountsClient.ui.Form} />
+const accountRoutes = () =>
```
Comment (alexyuly) [inline src/accountRoutes.js]: This is a component, so it's name should be Capitalized.

File: src/SignupFields.js (hunk 1 above)
Comment (TimMikeladze) [inline src/SignupFields.js]: Fixed.

File: src/SignupFields.js [hunk 3]
```diff
@@ -32,49 +32,44 @@ const SignupFields = ({ form, fields }) => {
...
           <AccountsClient.ui.PasswordField {...passwordProps({ form, fields })} />
           <AccountsClient.ui.PasswordConfirmField {...passwordConfirmProps({ form, fields })} />
         </div>
       );
-      break;
     default:
       break;
   }
-  return jsx;
+  return null;
```
Comment (alexyuly) [inline src/SignupFields.js]: Lol really this should go under the `default` case. At the end of the day I don't care.

Comment (TimMikeladze) [inline src/SignupFields.js]: I'll fix it.

Comment (alexyuly) [inline src/SignupFields.js]: Thank you!

**Pull request merged**